src/libraries/System.Diagnostics.EventLog/src/GenerateMessages.targets

```diff
@@ -0,0 +1,68 @@
+      <_eventLogMessagesAssemblyInfo>$(IntermediateOutputPath)$(_eventLogMessagesAssemblyName).AssemblyInfo.cs</_eventLogMessagesAssemblyInfo>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="GenerateEventLogMessages"
+          AfterTargets="CoreCompile"
+          DependsOnTargets="ResolveReferences;GenerateTargetFrameworkMonikerAttribute;GetAssemblyAttributes;GetEventLogMessagePaths"
+          Inputs="$(_eventLogMessagesResources);$(MSBuildAllProjects)"
+          Outputs="$(_eventLogMessagesOutputAssembly)">
+    <ItemGroup>
+      <_eventLogMessagesCompile Include="$(TargetFrameworkMonikerAssemblyAttributesPath)" />
+
+      <_eventLogMessagesAttribute Include="@(AssemblyAttribute)" />
+      <_eventLogMessagesAttribute Condition="'%(_eventLogMessagesAttribute._Parameter1)' == '$(AssemblyName)'">
+        <_Parameter1>$(_eventLogMessagesAssemblyName)</_Parameter1>
+      </_eventLogMessagesAttribute>
+      <!-- we don't require this attribute and don't want to define it on downlevel platforms -->
+      <_eventLogMessagesAttribute Remove="System.Runtime.Versioning.SupportedOSPlatform" />
+    </ItemGroup>
+
+    <WriteCodeFragment AssemblyAttributes="@(_eventLogMessagesAttribute)" Language="C#" OutputFile="$(_eventLogMessagesAssemblyInfo)">
+      <Output TaskParameter="OutputFile" ItemName="_eventLogMessagesCompile" />
+      <Output TaskParameter="OutputFile" ItemName="FileWrites" />
+    </WriteCodeFragment>
+
+    <Csc OutputAssembly="$(_eventLogMessagesOutputAssembly)"
+         Sources="@(_eventLogMessagesCompile)"
+         Win32Resource="$(_eventLogMessagesResources)"
+         References="@(ReferencePath)"
+         DebugType="$(DebugType)"
+         KeyContainer="$(KeyContainerName)"
+         KeyFile="$(KeyOriginatorFile)"
+         NoConfig="true"
+         NoLogo="$(NoLogo)"
+         NoStandardLib="$(NoCompilerStandardLib)"
+         PdbFile="$(_eventLogMessagesOutputSymbols)"
+         PublicSign="$(PublicSign)"
+         DelaySign="$(DelaySign)"
+         Deterministic="$(Deterministic)"
+         DisabledWarnings="$(DisabledWarnings)"
+         WarningLevel="$(WarningLevel)"
+         WarningsAsErrors="$(WarningsAsErrors)"
+         WarningsNotAsErrors="$(WarningsNotAsErrors)"
+         TargetType="Library"
+         ToolExe="$(CscToolExe)"
+         ToolPath="$(CscToolPath)"
+         UseSharedCompilation="$(UseSharedCompilation)">
+
+      <Output TaskParameter="OutputAssembly" ItemName="FileWrites"/>
+      <Output TaskParameter="OutputAssembly" ItemName="ReferenceCopyLocalPaths" />
+    </Csc>
+  </Target>
+  <Target Name="_AddMessagesToPackage"
+          BeforeTargets="GetFilesToPackage"
+          DependsOnTargets="GetEventLogMessagePaths">
+    <ItemGroup>
+      <AdditionalFileToPackage Include="$(_eventLogMessagesOutputAssembly)" />
+      <AdditionalFileToPackage Include="$(_eventLogMessagesOutputSymbols)" />
```

Comment:
What will be the targetPath for this Assembly  in the  package ? do we need to specify that. ?

Reply:
By placing it in AdditionalFileToPackage it is placed next to the assembly itself in lib/tfm.  That said, this will change once I refactor this to a separate project.

Reply:
Next to the primary output. https://github.com/dotnet/arcade/blob/866b2acd5ffc3c2031c102f6415fd7c6a1a370d5/src/Microsoft.DotNet.Build.Tasks.Packaging/src/build/PackageLibs.targets#L163-L167

That said, this will go away after I refactor to a project.

Reply:
I actually chose to keep the use of AdditionalFileToPackage since I think it does a nice job of ensuring we match the message dll to the place where we ship a real implementation, rather than requiring us to keep the configuration of the projects in sync.